src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -177,12 +184,43 @@ async def _run_impl(self) -> None:
                 else:
                     self._buffer.update(sample)
 
+                self.count_samples += 1
+                if self.wait_for_num_samples != 0:
+                    if self.count_samples % self.wait_for_num_samples == 0:
```

Comment:
You're using the same check in line 217, maybe put it in a common private function that way if you change one part it's also correct in the other part

Reply:
this check is gone